workdir/copy/link/build.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+comp=""
+plat=""
+arch=""
+fs=""
+extra=""
+
+name=$(basename "$0" .sh)
+
+echo "$name" | grep "^clang-" > /dev/null 2>&1
+if test $? -eq 0; then
+    comp="clang"
+else
+    echo "$name" | grep "^gcc-" > /dev/null 2>&1
+    if test $? -eq 0; then
+        comp="gcc"
+    else
+        echo "Unknown compiler" 1>&2
+        exit 1
+    fi
+fi
+
+echo "$name" | grep "\-qemu\-" > /dev/null 2>&1
+if test $? -eq 0; then
+    plat="qemu"
+else
+    echo "$name" | grep "\-fc\-" > /dev/null 2>&1
+    if test $? -eq 0; then
+        plat="fc"
+    else
+        echo "Unknown platform" 1>&2
+        exit 1
+    fi
+fi
+
+echo "$name" | grep "\-aarch64" > /dev/null 2>&1
+if test $? -eq 0; then
+    arch="aarch64"
+else
+    echo "$name" | grep "\-x86_64" > /dev/null 2>&1
+    if test $? -eq 0; then
+        arch="x86_64"
+    else
+        echo "Unknown architecture" 1>&2
+        exit 1
+    fi
+fi
+
+rest=$(echo "$name" | sed 's/^.*\-\(x86_64\|aarch64\)//')
+
+if test ! -z "$rest"; then
+    echo "$rest" | grep '^\-9pfs' > /dev/null 2>&1
+    if test $? -eq 0; then
+        fs="9pfs"
+    else
+        echo "$rest" | grep '^\-initrd' > /dev/null 2>&1
+        if test $? -eq 0; then
+            fs="initrd"
+        fi
+    fi
+fi
+
+if test -z "$fs"; then
+    extra="$rest"
+else
+    extra=$(echo "$rest" | sed 's/^\-[^-]\+//')
+fi
+
+echo "comp: $comp"
+echo "plat: $plat"
+echo "arch: $arch"
+echo "fs: $fs"
+echo "extra: $extra"
+
+config_suffix="$plat"-"$arch"
+if test ! -z "$fs"; then
+    config_suffix="$config_suffix"-"$fs"
+fi
+if test ! -z "$extra"; then
+    config_suffix="$config_suffix""$extra"
+fi
+
+# Do actual build steps.
+make distclean
+config=$(ls $(pwd)/.config.*-"$config_suffix")
+UK_DEFCONFIG=$config make defconfig
+make prepare
+if test "$comp" = "clang"; then
+    make CC=clang -j $(nproc)
+else
+    make -j $(nproc)
+fi
```

Comment:
```suggestion
make CC="$comp" -j $(nproc)
```
Besides being cleaner, I think we could use this in case we want to use different compiler versions, `$comp` might be something like `gcc-11`.

Reply:
You're right. I'll do that.